peer/interfaces.go

```diff
@@ -4,9 +4,36 @@ import (
 	"net"
 	"time"
 
+	"github.com/lightningnetwork/lnd/htlcswitch"
+	"github.com/lightningnetwork/lnd/lnwallet"
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
+// messageSwitch is an interface that abstracts managing the lifecycle of
+// abstract links. This is intended for peer package usage only.
+type messageSwitch interface {
+	// BestHeight returns the best height known to the messageSwitch.
+	BestHeight() uint32
+
+	// CircuitModifier returns a reference to the messageSwitch's internal
+	// CircuitModifier which abstracts the paths payments take and allows
+	// modifying them.
+	CircuitModifier() htlcswitch.CircuitModifier
+
+	// RemoveLink removes an abstract link given a ChannelID.
+	RemoveLink(cid lnwire.ChannelID)
+
+	// CreateAndAddLink creates an abstract link in the messageSwitch given
+	// a ChannelLinkConfig and raw LightningChannel pointer.
+	CreateAndAddLink(cfg htlcswitch.ChannelLinkConfig,
+		lnChan *lnwallet.LightningChannel) error
+
+	// GetLinksByInterface retrieves abstract links (represented by the
+	// ChannelUpdateHandler interface) based on the provided public key.
+	GetLinksByInterface(pub [33]byte) ([]htlcswitch.ChannelUpdateHandler,
```

Comment:
ok I see. Maybe we could add `GetLink` here too?

Reply:
I think if we're not using `GetLink` anymore, we don't need to include it in the interface